### PR TITLE
pump.io: git-2015-11-09 -> 1.0.0

### PR DESCRIPTION
**Pull request by rvl**
###### Motivation for this change

New upstream release with some security fixes. Release notes: https://github.com/e14n/pump.io/releases/tag/v1.0.0

Updated nix generation to use the new node2nix, so I ask @svanderburg to please have a look.

Please note: this won't solve the release tests failing on i686-linux. For that, #18574 and #18622 need to be merged. As I understand it, without these PRs (at least the first one), it's impossible to configure mongodb to work on i686.

If this PR is merged, then #18615 can be closed.

Cheers
###### Things done
- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [X] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/web-apps/pump.io/generate.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+curl https://raw.githubusercontent.com/e14n/pump.io/v1.0.0/package.json | \
+    jshon -e dependencies              \
+          -s '*' -i databank-mongodb   \
+          -s '*' -i databank-redis     \
+          -s '*' -i databank-memcached \
+          -s '*' -i databank-lrucache  \
+          -p | sed 's=\\/=/=g' > full-package.json
+
+node2nix --input full-package.json --composition composition.nix
+
```
Comment (svanderburg) [inline pkgs/servers/web-apps/pump.io/generate.sh]: Can you an additional parameter to node2nix?

Changing the invocation to this:

```bash
node2nix --input full-package.json --composition composition.nix --node-env ../../../development/node-packages/node-env.nix
```

makes it possible to reuse node-env from the NPM package set preventing a duplicate copy.

File: pkgs/servers/web-apps/pump.io/node-env.nix [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+      '';
+
+      # Provide the dependencies in a development shell through the NODE_PATH environment variable
+      inherit nodeDependencies;
+      shellHook = stdenv.lib.optionalString (dependencies != []) ''
+        export NODE_PATH=$nodeDependencies/lib/node_modules
+      '';
+    };
+in
+{ inherit buildNodeSourceDist buildNodePackage buildNodeShell; }
```
Comment (svanderburg) [inline pkgs/servers/web-apps/pump.io/node-env.nix]: We can git rid of a duplicate copy by referencing to the version in pkgs/development/node-packages

**Comment by rvl** [comment]
OK thanks @svanderburg, new version is pushed.

Do you know if there's anything better I can do about the extra database backend dependencies that this app has? If I simply list them in `node-packages.json` like `["pump.io", "databank-mongodb", "databank-redis", ...]`, they are not accessible by pump.io because they aren't within its `node_modules`.

**Comment by domenkozar** [comment]
@svanderburg all good now?

**Comment by rvl** [comment]
I think the only issue is that the package generation in `generate.sh` is not simple.
So I ask if anyone has any other objections to the change.

**Comment by Mic92** [comment]
thanks!

**Comment by rvl** [comment]
Thanks @Mic92. Can I ask that this commit also be cherry-picked onto the `release-16.09` branch because it contains security fixes?

**Comment by Mic92** [comment]
@rvl is https://github.com/NixOS/nixpkgs/pull/18574 and https://github.com/NixOS/nixpkgs/pull/18622 already in release-16.09 included?

**Comment by rvl** [comment]
@Mic92 Sorry I forgot to confirm this. Yes: they are, and this commit applies cleanly. Also the necessary node2nix changes exist in release-16.09.

**Comment by Mic92** [comment]
done in https://github.com/NixOS/nixpkgs/commit/246c9375996e3bf5af8c1f2f8179390ee63a950a

**Pull request merged**